[CI] FiltersAggregatorTests.testMatchNoneTopLevel failing on master

This happend on a local run but seems reproducable on master

**Build scan**:

Edit: this is from a CI test failure on Sunday that looks similar
https://gradle-enterprise.elastic.co/s/wphgk3nxj73jg

**Repro line**:

```
./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.testMatchNoneTopLevel" -Dtests.seed=D315C18B9D436BD4 -Dtests.locale=zh-TW -Dtests.timezone=Africa/Timbuktu -Druntime.java=16
```

**Reproduces locally?**:

yes

**Applicable branches**:

master

**Failure history**:

**Failure excerpt**:

```
org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests > testMatchNoneTopLevel FAILED
    java.lang.AssertionError: Expected a map containing
                           test: a map containing
    segments_with_doc_count_field: <0>
       segments_with_deleted_docs: <0>
               segments_collected: <0>
                 segments_counted: expected <1> but was <2>
                          filters: a list containing
                                  0: a map containing
                                query: "MatchNoDocsQuery(\"User requested \"match_none\" query.\")"
                      specialized_for: "match_none"
        at __randomizedtesting.SeedInfo.seed([D315C18B9D436BD4:1A0FCEB66D3AF1FE]:0)
        at io.github.nik9000.mapmatcher.MapMatcher.assertMap(MapMatcher.java:78)
        at io.github.nik9000.mapmatcher.MapMatcher.assertMap(MapMatcher.java:61)
        at org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.lambda$testMatchNoneTopLevel$25(FiltersAggregatorTests.java:678)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:634)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.lambda$debugTestCase$9(AggregatorTestCase.java:589)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.withIndex(AggregatorTestCase.java:553)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:589)
        at org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.testMatchNoneTopLevel(FiltersAggregatorTests.java:669)
```

A different test in the same test suite failed today with a similar error: https://gradle-enterprise.elastic.co/s/vypetlhquslnk

```

org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests > testTermFilter FAILED
    java.lang.AssertionError: Expected a map containing
                           test: a map containing
    segments_with_doc_count_field: <0>
       segments_with_deleted_docs: <0>
               segments_collected: <0>
                 segments_counted: expected <1> but was <2>
                          filters: a list containing
                                  0: a map containing
                                query: "f:0"
                      specialized_for: "term"
                results_from_metadata: <2>
        at __randomizedtesting.SeedInfo.seed([4A87550F20633125:583542EB712611B0]:0)
        at io.github.nik9000.mapmatcher.MapMatcher.assertMap(MapMatcher.java:78)
        at io.github.nik9000.mapmatcher.MapMatcher.assertMap(MapMatcher.java:61)
        at org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.lambda$testTermFilter$27(FiltersAggregatorTests.java:717)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:634)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.lambda$debugTestCase$9(AggregatorTestCase.java:589)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.withIndex(AggregatorTestCase.java:553)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:589)
        at org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.testTermFilter(FiltersAggregatorTests.java:708)
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.testTermFilter" -Dtests.seed=4A87550F20633125 -Dtests.locale=uk -Dtests.timezone=Africa/Banjul -Druntime.java=11

```

--- a/server/src/test/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregatorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregatorTests.java
@@ -89,6 +89,7 @@ import static io.github.nik9000.mapmatcher.MapMatcher.assertMap;
 import static io.github.nik9000.mapmatcher.MapMatcher.matchesMap;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.nullValue;
@@ -411,7 +412,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                     matchesMap().entry("segments_with_doc_count_field", 0)
                         .entry("segments_with_deleted_docs", 0)
                         .entry("segments_collected", 0)
-                        .entry("segments_counted", 1)
+                        .entry("segments_counted", greaterThanOrEqualTo(1))
                         .entry("filters", matchesList().item(matchesMap().entry("query", "test:[1577836800000 TO 1583020799999]")))
                 )
             );
@@ -471,7 +472,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                     debug,
                     matchesMap().entry(
                         "test",
-                        matchesMap().entry("segments_counted", 1)
+                        matchesMap().entry("segments_counted", greaterThanOrEqualTo(1))
                             .entry("segments_collected", 0)
                             .entry("segments_with_doc_count_field", 0)
                             .entry("segments_with_deleted_docs", 0)
@@ -513,9 +514,9 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                     debug,
                     matchesMap().entry(
                         "test",
-                        matchesMap().entry("segments_counted", 1)
+                        matchesMap().entry("segments_counted", greaterThanOrEqualTo(1))
                             .entry("segments_collected", 0)
-                            .entry("segments_with_doc_count_field", 1)
+                            .entry("segments_with_doc_count_field", greaterThanOrEqualTo(1))
                             .entry("segments_with_deleted_docs", 0)
                             .entry(
                                 "filters",
@@ -583,7 +584,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                 aggregator.collectDebugInfo(debug::put);
                 assertMap(
                     debug,
-                    matchesMap().entry("segments_counted", 1)
+                    matchesMap().entry("segments_counted", greaterThanOrEqualTo(1))
                         .entry("segments_collected", 0)
                         .entry("segments_with_doc_count_field", 0)
                         .entry("segments_with_deleted_docs", 0)
@@ -670,7 +671,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                         matchesMap().entry("segments_with_doc_count_field", 0)
                             .entry("segments_with_deleted_docs", 0)
                             .entry("segments_collected", 0)
-                            .entry("segments_counted", 1)
+                            .entry("segments_counted", greaterThanOrEqualTo(1))
                             .entry(
                                 "filters",
                                 matchesList().item(
@@ -707,7 +708,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                         matchesMap().entry("segments_with_doc_count_field", 0)
                             .entry("segments_with_deleted_docs", 0)
                             .entry("segments_collected", 0)
-                            .entry("segments_counted", 1)
+                            .entry("segments_counted", greaterThanOrEqualTo(1))
                             .entry(
                                 "filters",
                                 matchesList().item(
@@ -746,7 +747,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                         matchesMap().entry("segments_with_doc_count_field", 0)
                             .entry("segments_with_deleted_docs", 0)
                             .entry("segments_collected", 0)
-                            .entry("segments_counted", 1)
+                            .entry("segments_counted", greaterThanOrEqualTo(1))
                             .entry(
                                 "filters",
                                 matchesList().item(
@@ -787,7 +788,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                         matchesMap().entry("segments_with_doc_count_field", 0)
                             .entry("segments_with_deleted_docs", 0)
                             .entry("segments_collected", 0)
-                            .entry("segments_counted", 1)
+                            .entry("segments_counted", greaterThanOrEqualTo(1))
                             .entry(
                                 "filters",
                                 matchesList().item(
@@ -874,7 +875,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                         "test",
                         matchesMap().entry("segments_with_doc_count_field", 0)
                             .entry("segments_with_deleted_docs", 0)
-                            .entry("segments_collected", 1)
+                            .entry("segments_collected", greaterThanOrEqualTo(1))
                             .entry("segments_counted", 0)
                             .entry(
                                 "filters",
@@ -952,7 +953,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                         "test",
                         matchesMap().entry("segments_with_doc_count_field", 0)
                             .entry("segments_with_deleted_docs", 0)
-                            .entry("segments_collected", 1)
+                            .entry("segments_collected", greaterThanOrEqualTo(1))
                             .entry("segments_counted", 0)
                             .entry("filters", hasSize(2))
                     ).entry("test.s", matchesMap()).entry("test.m", matchesMap())
@@ -1033,7 +1034,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                         "test",
                         matchesMap().entry("segments_with_doc_count_field", 0)
                             .entry("segments_with_deleted_docs", 0)
-                            .entry("segments_collected", 1)
+                            .entry("segments_collected", greaterThanOrEqualTo(1))
                             .entry("segments_counted", 0)
                             .entry("filters", hasSize(buckets.size()))
                     ).entry("test.s", matchesMap()).entry("test.m", matchesMap())

